### PR TITLE
#1116 refactor: remove `pluralized` and add `singular`

### PR DESCRIPTION
**Pull request by stefanorosanelli**
This PR is a refactor of `object_types`:
 * `name` is plural, since actual type names are used in plural form
 * `singular` is the singular version of a type name (used in entities and elsewhere)
 * `pluralized` was removed

### CODE REVIEW

File: plugins/BEdita/Core/config/Migrations/20170301144829_RemovePluralized.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                'null' => true,
+            ])
+            ->update();
+
+        $this->query('UPDATE object_types set singular = name');
+        $this->query('UPDATE object_types set name = pluralized');
+
+        $this->table('object_types')
+            ->changeColumn('singular', 'string', ['null' => false])
+            ->update();
```
Comment (fquffio) [inline plugins/BEdita/Core/config/Migrations/20170301144829_RemovePluralized.php]: Why not `$this->table('object_types')->renameColumn('name', 'singular')->update()`, then `$this->table('object_types')->renameColumn('pluralized', 'name')->update()` ?

File: plugins/BEdita/Core/tests/TestCase/Model/Entity/ObjectTypeTest.php [hunk 2]
```diff
@@ -135,39 +135,39 @@ public function testSetName()
...
      * @param string $name Object type name.
-     * @param string|null $pluralized Object type pluralized name.
+     * @param string|null $singular Object type singular name.
      * @return void
      *
-     * @dataProvider getSetPluralizedProvider
-     * @covers ::_getPluralized()
-     * @covers ::_setPluralized()
+     * @dataProvider getSetSingularProvider
+     * @covers ::_getsingular()
```
Comment (fquffio) [inline plugins/BEdita/Core/tests/TestCase/Model/Entity/ObjectTypeTest.php]: It should probably be `_getSingular()`. I'm not sure whether casing does matter to PHPUnit…

Same problem on next line.

File: plugins/BEdita/Core/tests/TestCase/Model/Entity/ObjectTypeTest.php [hunk 3]
```diff
@@ -190,7 +190,7 @@ public function testGetAlias()
     }
 
     /**
-     * Data provider for `testGetSetPluralized` test case.
+     * Data provider for `testGetSetsingular` test case.
```
Comment (fquffio) [inline plugins/BEdita/Core/tests/TestCase/Model/Entity/ObjectTypeTest.php]: This is a data provider for `testGetSetTable`. The comment was wrong before, but maybe we can take the chance to fix.

**Review by fquffio — COMMENTED** [review]
Minor annotations as usual…

File: plugins/BEdita/Core/config/Migrations/20170301144829_RemovePluralized.php (hunk 1 above)
Comment (stefanorosanelli) [reply plugins/BEdita/Core/config/Migrations/20170301144829_RemovePluralized.php]: much better, I found some issues when I first tested it that way, but somehow I was wrong, it works!

File: plugins/BEdita/Core/tests/TestCase/Model/Entity/ObjectTypeTest.php (hunk 2 above)
Comment (stefanorosanelli) [reply plugins/BEdita/Core/tests/TestCase/Model/Entity/ObjectTypeTest.php]: yesss

File: plugins/BEdita/Core/tests/TestCase/Model/Entity/ObjectTypeTest.php (hunk 3 above)
Comment (stefanorosanelli) [reply plugins/BEdita/Core/tests/TestCase/Model/Entity/ObjectTypeTest.php]: :+1:

**Pull request merged**